Fabric.Identity.API/Management/UsersController.cs

```diff
@@ -102,26 +102,21 @@ public IActionResult Search(string searchText, string identityProvider)
 
         private async Task<IActionResult> ProcessSearchRequest(string clientId, IEnumerable<string> userIds)
         {
-            if (string.IsNullOrEmpty(clientId))
-            {
-                return CreateFailureResponse(
-                    "No client id was included in the request. Please specify a client id",
-                    HttpStatusCode.BadRequest);
-            }
-
             var docIds = userIds.Select(id => id?.ToLower()).ToList();
             if (!docIds.Any() || docIds.All(string.IsNullOrEmpty))
             {
                 return CreateFailureResponse("No userIds were included in the request",
                     HttpStatusCode.BadRequest);
             }
 
-            var client = _clientManagementStore.FindClientByIdAsync(clientId).Result;
-
-            if (string.IsNullOrEmpty(client?.ClientId))
+            if (!string.IsNullOrEmpty(clientId))
             {
-                return CreateFailureResponse($"The specified client with id: {clientId} was not found",
-                    HttpStatusCode.NotFound);
+                var client = _clientManagementStore.FindClientByIdAsync(clientId).Result;
+                if (string.IsNullOrEmpty(client?.ClientId))
+                {
+                    return CreateFailureResponse($"The specified client with id: {clientId} was not found",
+                        HttpStatusCode.NotFound);
+                }

```

Comment:
What happens in line 124 when clientId is null or empty when we call ToUserViewModel?

Reply:
A null `LastLoginDate` is returned. Below is the implementation.

```
 public static UserApiModel ToUserViewModel(this User user, string clientId)
        {
            var lastLoginForClient = user.LastLoginDatesByClient
                .SingleOrDefault(l => l.ClientId.Equals(clientId, StringComparison.OrdinalIgnoreCase));

            DateTime? dateToSet = lastLoginForClient?.LoginDate;

            return new UserApiModel
            {
                SubjectId = user.SubjectId,
                FirstName = user.FirstName,
                LastName = user.LastName,
                MiddleName = user.MiddleName,
                LastLoginDate = dateToSet
            };
        }
```

Reply:
I'll add a check to the beginning of that method so we bypass all of the "last login date" logic if `clientId` is null or empty.

Reply:
Thanks.